lib/vector-core/src/tcp.rs

```diff
@@ -8,6 +8,7 @@ use vector_config::configurable_component;
 #[serde(deny_unknown_fields)]
 pub struct TcpKeepaliveConfig {
     /// The time to wait, in seconds, before starting to send TCP keepalive probes on an idle connection.
+    #[configurable(metadata(docs::type_unit = "seconds"))]
     pub time_secs: Option<u64>,
```

Comment:
Did you try changing this to a `Duration`?

Reply:
I did. The issue is that near as I could tell there isn't a standard default we are using, and when I converted it to `Option<Duration>`, I saw compilation issues so turned around. But, is `Option<Duration>` desirable? I can give it another look.

Reply:
👍 there were problems with it before, I wasn't sure if more recent changes Toby has merged.

Reply:
@spencergilbert The problem with `Option<T>` and `serde_with` helpers is still present, AFAIK. I did fix another problem with missing metadata specifically for `Option<T>`, which is what you might be thinking of.

Reply:
Yeah that sounds right